Index: script.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- script.js	(date 1619877816281)
+++ script.js	(date 1619877816281)
@@ -13,82 +13,101 @@
             let raw_name = name.replace(/\.[^/.]+$/, "")
             const poster = `thumbnails/${raw_name}.png`
             html = `<a  href="#" id=${path}><img class="video" src="${poster}" alt="" width="220" height="220"></a>`;
-            div.insertAdjacentHTML('afterend',html);
+            div.insertAdjacentHTML('afterend', html);
         })
     });
     doVideoLogic();
 }
 
-function dragElement(elmnt) {
-    let scale = 1;
-    let pos1 = 0, pos2 = 0, pos3 = 0, pos4 = 0;
+
+
+function hi () {alert(1)}
+
+function doVideoLogic() {
+    const modal = document.querySelector('.modal');
+    const videos = document.querySelectorAll('.video');
+    const video = document.querySelector('.modal video');
+    function dragAndZoom(elmnt) {
+        let scale = 1;
+        let pos1 = 0, pos2 = 0, pos3 = 0, pos4 = 0;
+        let start, end, delta
         elmnt.onmousedown = dragMouseDown;
         elmnt.onwheel = zoom;
 
-    function zoom(e) {
-        e = e || window.event;
-        e.preventDefault();
-        let delta = e.deltaY
-        if (delta > 0) {
-            if(scale < 0.50) {return}
-            scale -= 0.25
-        } else {
-            scale += 0.25
-        }
-        elmnt.style.transform = `scale(${scale})`;
-    }
+        function zoom(e) {
+            e = e || window.event;
+            e.preventDefault();
+            let delta = e.deltaY
+            if (delta > 0) {
+                if(scale < 0.50) {return}
+                scale -= 0.25
+            } else {
+                scale += 0.25
+            }
+            elmnt.style.transform = `scale(${scale})`;
+        }
 
-    function dragMouseDown(e) {
-        e = e || window.event;
-        e.preventDefault();
-        // get the mouse cursor position at startup:
-        pos3 = e.clientX;
-        pos4 = e.clientY;
-        document.onmouseup = closeDragElement;
-        // call a function whenever the cursor moves:
-        document.onmousemove = elementDrag;
-    }
+        function dragMouseDown(e) {
+            start = new Date();
+            e = e || window.event;
+            e.preventDefault();
+            // get the mouse cursor position at startup:
+            pos3 = e.clientX;
+            pos4 = e.clientY;
+            document.onmouseup = closeDragElement;
+            // call a function whenever the cursor moves:
+            document.onmousemove = elementDrag;
+        }
 
-    function elementDrag(e) {
-        e = e || window.event;
-        e.preventDefault();
-        // calculate the new cursor position:
-        pos1 = pos3 - e.clientX;
-        pos2 = pos4 - e.clientY;
-        pos3 = e.clientX;
-        pos4 = e.clientY;
-        // set the element's new position:
-        elmnt.style.top = (elmnt.offsetTop - pos2) + "px";
-        elmnt.style.left = (elmnt.offsetLeft - pos1) + "px";
-    }
+        function elementDrag(e) {
+            e = e || window.event;
+            e.preventDefault();
+            // calculate the new cursor position:
+            pos1 = pos3 - e.clientX;
+            pos2 = pos4 - e.clientY;
+            pos3 = e.clientX;
+            pos4 = e.clientY;
+            // set the element's new position:
+            elmnt.style.top = (elmnt.offsetTop - pos2) + "px";
+            elmnt.style.left = (elmnt.offsetLeft - pos1) + "px";
+        }
 
-    function closeDragElement() {
-        // stop moving when mouse button is released:
-        document.onmouseup = null;
-        document.onmousemove = null;
-    }
-}
-function resizeByScroll(e) {
-    console.log(e)
-
-}
-
-
-function hi () {alert(1)}
-
-function doVideoLogic() {
-    const modal = document.querySelector('.modal');
-    const videos = document.querySelectorAll('.video')
-    function openVideo (e) {
+        function closeDragElement(e) {
+            end = new Date();
+            delta = (end - start) / 1000.0;
+            if (delta < 0.1) {
+                video.pause();
+                modal.classList.remove('show');
+            } else{
+                // stop moving when mouse button is released:
+                document.onmouseup = null;
+                document.onmousemove = null;
+            }
+        }
+    }
+    function openVideo (e) {
+        console.log(e)
         const path = e.path[1]['id'];
-        document.querySelector('.modal video').src =`${path}`;
+        video.src =`${path}`;
         modal.classList.add('show');
     }
+    function closeVideo (e) {
+        modal.addEventListener("mousedown", function(){
+
+        })
+
+        modal.addEventListener("mouseup", function() {
+
+        });
+    }
+
+
+    // Add listeners
     videos.forEach((video) => {
         video.addEventListener('click', openVideo);
     });
-
-    dragElement(modal);
+    dragAndZoom(modal);
+    closeVideo();
 }
 
 eel.get_sorted_webm()(createPage);
